server/src/test/java/org/elasticsearch/cluster/coordination/NodeJoinExecutorTests.java

```diff
@@ -227,6 +230,227 @@ public Set<NodeFeature> getFeatures() {
         );
     }
 
+    @SuppressForbidden(reason = "we need to actually check what is in cluster state")
+    private static Map<String, Set<String>> getRecordedNodeFeatures(ClusterState state) {
+        return state.clusterFeatures().nodeFeatures();
+    }
+
+    private static Version nextMajor() {
+        return Version.fromId((Version.CURRENT.major + 1) * 1_000_000 + 99);
```

Comment:
Until the assert in `VersionInformation` constructor is removed, we need to continue to use `Version` here